File: web_shop/src/main/webapp/js/controller/ordersController.js
app.controller('ordersController' ,function($scope,$controller,ordersService) {

    $controller('baseController', {$scope: $scope});//继承


    // 发货的方法:
    $scope.updateStatus = function(status){
        ordersService.updateStatus($scope.selectIds,status).success(function(response){
            if(response.success){
                $scope.reloadList();//刷新列表
                $scope.selectIds = [];
            }else{
                alert(response.message);
            }
        });
    }

    $scope.searchEntity={};//定义搜索对象

    // 显示状态
    $scope.status = ["","未付款","已付款","未发货","已发货","交易成功","交易关闭","待评价"];
    // 显示订单来源
    $scope.sourceType = ["app端","pc端","M端","微信端","手机qq端"];
    //分页搜索
    $scope.search=function(page,rows){
        ordersService.search(page,rows,$scope.searchEntity).success(
            function(response){
                 $scope.list=response.rows;
                $scope.paginationConf.totalItems=response.total;//更新总记录数
            }
        );
    }
});

File: web_shop/src/main/java/cn/itcast/core/controller/ordersController.java
package cn.itcast.core.controller;

import cn.itcast.core.pojo.entity.PageResult;
import cn.itcast.core.pojo.entity.Result;
import cn.itcast.core.pojo.order.Order;
import cn.itcast.core.service.OrdersService;
import com.alibaba.dubbo.config.annotation.Reference;
import org.springframework.security.core.context.SecurityContextHolder;
import org.springframework.web.bind.annotation.RequestBody;
import org.springframework.web.bind.annotation.RequestMapping;
import org.springframework.web.bind.annotation.RestController;

@RestController
@RequestMapping("/orders")
public class ordersController {
    @Reference
    private OrdersService ordersService;

    @RequestMapping("/search")
    public PageResult search(@RequestBody Order order, Integer page, Integer rows) {
        //获取当前登录用户用户名
        String userName = SecurityContextHolder.getContext().getAuthentication().getName();
        //将当前用户名放入卖家id字段
        order.setSellerId(userName);

        PageResult result = ordersService.findPage(order, page, rows);
        return result;
    }
    /*商品发货*/
    @RequestMapping("/updateStatus")
    public  Result updateStatus(Long[] selectIds, String status) {
        try {
            if (selectIds != null) {
                for (Long orderId : selectIds) {
                    Order order=new Order();
                    order.setOrderId(orderId);
                    //1. 更改数据库中商品的审核状态
                    ordersService.updateStatus(order, status);
                }
            }
            return new Result(true, "状态修改成功!");
        } catch (Exception e) {
            e.printStackTrace();
            return new Result(false, "状态修改失败!");
        }
    }

}
